fontforge/python.c

```diff
@@ -14066,6 +14066,150 @@ return (NULL);
+                return( -1 );
+            }
+        } else {
+            PyErr_Format(PyExc_TypeError, "The first element of a style set name specification must be a string or an integer.");
+            return( -1 );
+        }
+
+        PyObject *tag_str_py = PyTuple_GetItem(ss_names_tuple, 1);
+        if ( !PyUnicode_Check(tag_str_py) ) {
+            PyErr_Format(PyExc_TypeError, "The second element of a style set name specification must be a string.");
+            return( -1 );
+        }
+        tag_str = copy(PyUnicode_AsUTF8(tag_str_py));
+        is_valid = 0;
+        if ( strlen(tag_str)== 4 && tag_str[0]=='s' && tag_str[1]=='s'
+            && '0'<=tag_str[2] && tag_str[2]<='9' && '0'<=tag_str[3] && tag_str[3]<='9'
+        ) {
+            int ss_num = 10*(tag_str[2]-'0') + tag_str[3]-'0';
+            is_valid = 1 <= ss_num && ss_num <= 20;
+        }
+        if ( !is_valid ) {
+            PyErr_Format(PyExc_TypeError, "The second element of a style set name specification must be a valid style set tag.");
+            return( -1 );
+        }
+
+        if ( !PyUnicode_Check(PyTuple_GetItem(ss_names_tuple, 2)) ) {
+            PyErr_Format(PyExc_TypeError, "The third element of a style set name specification must be a string.");
+            return( -1 );
+        }
+    }
+
+    /* check for duplicates */
+    for ( i=0; i<rows; ++i ) {
+        for ( j=i+1; j<rows; ++j ) {
+            if ( PyTuple_GetItem(PyTuple_GetItem(value, i), 0) == PyTuple_GetItem(PyTuple_GetItem(value, j), 0) &&
+                PyTuple_GetItem(PyTuple_GetItem(value, i), 1) == PyTuple_GetItem(PyTuple_GetItem(value, j), 1)
+            ) {
+                PyErr_Format(PyExc_ValueError, "The feature %s is named twice in language %s:\n  %.80s\n  %.80s",
+                    PyUnicode_AsUTF8(PyTuple_GetItem(PyTuple_GetItem(value, i), 1)),
+                    PyUnicode_AsUTF8(PyTuple_GetItem(PyTuple_GetItem(value, i), 0)),
+                    PyUnicode_AsUTF8(PyTuple_GetItem(PyTuple_GetItem(value, i), 2)),
+                    PyUnicode_AsUTF8(PyTuple_GetItem(PyTuple_GetItem(value, j), 2))
+                    );
+                return( -1 );
+            }
+        }
+    }
+
+    /* ok, this is valid input */
+    OtfFeatNameListFree(sf->feat_names);
+    sf->feat_names = NULL;
+    for ( i=rows-1; i>=0; --i ) {
+        ss_names_tuple = PyTuple_GetItem(value, i);
+
+        lang_py = PyTuple_GetItem(ss_names_tuple, 0);
+        if ( PyUnicode_Check(lang_py) ) {
+            lang = FindFlagByName(sfnt_name_mslangs, PyUnicode_AsUTF8(lang_py));
+        } else {
+            lang = PyLong_AsLong(lang_py);
+        }
+
+        tag = StrObjToTag(PyTuple_GetItem(ss_names_tuple, 1), false);
+
+        name_str = copy(PyUnicode_AsUTF8(PyTuple_GetItem(ss_names_tuple, 2)));
+
+        for ( fn=sf->feat_names; fn!=NULL && fn->tag!=tag; fn=fn->next );
```

Comment:
Note that this code appears to use the existing `feat_names` entry for the tag, or (below) adds a new entry for a missing tag.